blivet/devicelibs/edd.py

```diff
@@ -32,66 +32,193 @@
+# bug, but it's part of the ABI now, so it sticks.  The format of the
+# scnprintf() is at least correct.
+re_interface_raid = re.compile(r'^RAID\s*identity_tag: (\S*)\s*$')
+re_interface_edd3_sata = re.compile(r'^SATA\s*device: (\S*)\s*$')
+# EDD 4 features from 2010 and later.  Awesomely, the "version" output from
+# int 13 AH=41h says: AH Version of extensions. Shall be set to 30h,
+# so there's no way to distinguish these from EDD 3, even thuogh SATA does
+# differ.  In theory, if we're on <4.0, pmp should always be all 0's.
+re_interface_edd4_sata = re.compile(r'^SATA\s*device: (\S*)\s*pmp: (\S*)\s*$')
+re_interface_sas = re.compile(r'^SAS\s*sas_address: (\S*)\s*lun: \(\S*\)\s*$')
+# to make life difficult, when it finds an unknown interface type string,
+# the kernel prints the values without the string.  But it does print the
+# anchoring tab that would go between them...
+re_interface_unknown = re.compile(r'^(\S*)\s*unknown: (\S*) (\S*)\s*$')
 
 class EddEntry(object):
     """ This object merely collects what the /sys/firmware/edd/* entries can
         provide.
     """
     def __init__(self, sysfspath):
+        # the path we're probing
+        self.sysfspath = sysfspath
+
+        # data used for logging later
+        self.interface = None
+
+        # data from edd/int13_devXX/version
+        self.version = util.get_sysfs_attr(sysfspath, "version")
+
+        # data from edd/int13_devXX/mbr_signature
+        self.mbr_sig = None
+
+        # data from edd/int13_devXX/sectors
+        self.sectors = None
+
+        # data from edd/int13_devXX/host_bus
+        self.host_bus = None
+        self.pci_dev = None
+        self.channel = None
+
+        # data from edd/int13_devXX/interface in (nearly) all cases
         self.type = None
 
+        # data from edd/int13_devXX/interface when type is ATAPI
+        self.atapi_device = None
+        self.atapi_lun = None
+
+        # data from edd/int13_devXX/interface when type is ATA or SATA
         self.ata_device = None
-        self.channel = None
-        self.mbr_signature = None
-        self.pci_dev = None
+        self.ata_pmp = None
+
+        # data from edd/int13_devXX/interface when type is SCSI
         self.scsi_id = None
         self.scsi_lun = None
-        self.sectors = None
 
-        self.load(sysfspath)
+        # data from edd/int13_devXX/interface when type is USB
+        self.usb_serial = None
+
+        # data from edd/int13_devXX/interface when type is 1394
+        self.ieee1394_eui64 = None
+
+        # data from edd/int13_devXX/interface when type is FIBRE
+        self.fibre_wwid = None
+        self.fibre_lun = None
+
+        # data from edd/int13_devXX/interface when type is I2O
+        self.i2o_identity = None
+
+        # data from edd/int13_devXX/interface when type is SAS
+        self.sas_address = None
+        self.sas_lun = None
+
+        # data we may get as the result of the search
+        self.sysfslink = None
```

Comment:
All these descriptions in comments could be turned into an actual documentation by changing the comments to be docstrings (`"""`strings right below the attributes)